lib/pbench/server/cache_manager.py

```diff
@@ -430,29 +429,61 @@ def get_info(self, path: Path) -> JSONOBJECT:
         return fd_info
 
     @staticmethod
-    def extract(tarball_path: Path, path: str) -> Optional[str]:
-        """Extract a file from the tarball and return it as a string
+    def filestream(tarball_path: Path, path: Path) -> Optional[IO[bytes]]:
+        """Extracts filestream from the tarball
+
+        Reports failure by raising Exceptions
```

Comment:
I think you can omit this line, since the "Raise` section at line 440 indicates the same thing and provides better detail.